lib/checks/navigation/identical-links-same-purpose-after.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Skip unless there are more than a single result
+ */
+if (results.length < 2) {
+	return results;
+}
+
+/**
+ * for each result
+ * - get other results with matching accessible name
+ * - check if same purpose is served
+ *  - if not change `result` to `undefined`
+ */
+for (let index = 0; index < results.length; index++) {
+	const { data: nodeData } = results[index];
+	const { name, resource } = nodeData;
+	const identicalNodes = getNodesWithIdenticalAccessibleName(name, index);
+	if (!identicalNodes.length) {
+		continue;
+	}
+	if (!resource) {
+		results[index].result = undefined;
+		continue;
+	}
+	if (!servesSamePurpose(resource, identicalNodes)) {
+		results[index].result = undefined;
+		continue;
+	}
+}
+
+return results;
+
+/**
+ * Get list of nodes from results which match a given accessible name
+ * @method getNodesWithIdenticalAccessibleName
+ * @param {String} expectedName accessible name to be matched
+ * @param {Number} excludeIndex exclude `index` of result, that should not be taken into consideration
+ * @returns {Array<Object>}
+ */
+function getNodesWithIdenticalAccessibleName(expectedName, excludeIndex) {
+	return results.filter((result, index) => {
+		const {
+			data: { name }
+		} = result;
+		return index !== excludeIndex && name === expectedName;
+	});
+}
+
+/**
+ * Check if a given set of nodes have same resource
+ * @method servesSamePurpose
+ * @param {String} expectedResource expected resource
+ * @param {Array<Object>} identicalNodes results where `data.accessibleText` were identical
+ * @returns {Boolean}
+ */
+function servesSamePurpose(expectedResource, identicalNodes) {
+	const { parts: expected } = parseUri(expectedResource);
+	return identicalNodes.every(({ data }) => {
+		const { resource = undefined } = data;
+		if (!resource) {
+			return false;
+		}
+		const { isFile, parts: actual } = parseUri(resource);
+		return actual[isFile ? 'some' : 'every'](part => expected.includes(part));
+	});
+}
+
+/**
+ * Parse a given URI and return to parts
+ * @method parseUri
+ * @param {String} uri uri resource
+ * @returns {Object}
+ */
+function parseUri(uri = '') {
+	const parser = document.createElement('a');
+	parser.href = uri;
+
+	const curatedPathname = stripLeadingAndTrailingSlash(parser.pathname);
+	const defaults = [
+		parser.protocol,
+		parser.hostname,
+		parser.port,
+		parser.search
+	];
+	const pathname = parseUriPathnameAndHash(curatedPathname, parser.hash);
+	const file = parseUriPathnameAndFile(curatedPathname);
+
+	let uriParts = [...defaults, ...pathname];
+	if (file) {
+		uriParts = [...defaults, ...file];
```

Comment:
It took me 5 reads to realise you were overriding uriParts rather than concatenating `file` to it. You have forking logic here. Either file, or something else. That's what `if / else` is for. Not using `if / else` here sets incorrect expectations, and lead to someone having to read this code 5 times before they understand what's happening. You're also using memory that shouldn't be, because you create an array and than immediately override it.